lib/plugin.js

```diff
@@ -0,0 +1,18 @@
+module.exports = function pluginMethod (plugin) {
+  var self = this
+  var p
+
+  if (typeof plugin === 'function') {
+    plugin(self)
+  }
+
+  if (typeof plugin === 'object') {
+    for (p in plugin) {
+      if (plugin.hasOwnProperty(p)) {
+        self[p] = plugin[p]
+      }
+    }
```

Comment:
Can you do this instead?

```js
Object.keys(plugin).forEach(function (key) {
  self[key] = plugin[key]
})
```

I actually don’t know which one performs better, but this version is simpler to test. Right now, we don’t have a test where `plugin.hasOwnProperty(p)` returns false, and we’d need to extend the Prototype for it which is kinda hacky